core/src/main/scala/kafka/coordinator/transaction/TransactionLog.scala

```diff
@@ -79,7 +79,7 @@ object TransactionLog {
     // Serialize with version 0 (highest non-flexible version) until transaction.version 1 is enabled
     // which enables flexible fields in records.
     val version: Short =
-      if (usesFlexibleRecords) 1 else 0
+      if (transactionVersionLevel >= 1) 1 else 0
```

Comment:
If we pass `TransactionVersion` to this method too, we could have a `recordVersion` which returns the version to use here.